digitaltwin_client/src/dt_client_core.c

```diff
@@ -854,7 +854,14 @@ static DIGITALTWIN_CLIENT_RESULT VerifyComponentsUnique(DIGITALTWIN_INTERFACE_CL
 
 // DT_ClientCoreRegisterInterfacesAsync updates the list of interfaces we're supporting and begins 
 // protocol update of server.
-DIGITALTWIN_CLIENT_RESULT DT_ClientCoreRegisterInterfacesAsync(DT_CLIENT_CORE_HANDLE dtClientCoreHandle, const char* deviceCapabilityModel, DIGITALTWIN_INTERFACE_CLIENT_HANDLE* dtInterfaces, unsigned int numDTInterfaces, DIGITALTWIN_INTERFACE_REGISTERED_CALLBACK dtInterfaceRegisteredCallback, void* userContextCallback)
+DIGITALTWIN_CLIENT_RESULT DT_ClientCoreRegisterInterfacesAsync(
```

Comment:
define behavior for DIGITALTWIN_NO_DEFAULT_COMPONENT and check for invalid dtDefaultInterfaceIndex

Reply:
default component is not going in for part 1 fix